impl/task_runner_do.go

```diff
@@ -23,110 +23,112 @@ import (
 )
 
 // NewTaskRunner creates a TaskRunner instance based on the task type.
-func NewTaskRunner(taskName string, task model.Task, taskSupport TaskSupport) (TaskRunner, error) {
+func NewTaskRunner(taskName string, task model.Task, workflowDef *model.Workflow) (TaskRunner, error) {
 	switch t := task.(type) {
 	case *model.SetTask:
-		return NewSetTaskRunner(taskName, t, taskSupport)
+		return NewSetTaskRunner(taskName, t)
 	case *model.RaiseTask:
-		return NewRaiseTaskRunner(taskName, t, taskSupport)
+		return NewRaiseTaskRunner(taskName, t, workflowDef)
 	case *model.DoTask:
-		return NewDoTaskRunner(t.Do, taskSupport)
+		return NewDoTaskRunner(t.Do, workflowDef)
 	case *model.ForTask:
-		return NewForTaskRunner(taskName, t, taskSupport)
+		return NewForTaskRunner(taskName, t, workflowDef)
+	case *model.CallHTTP:
+		return NewCallHttpRunner(taskName, t)
 	default:
 		return nil, fmt.Errorf("unsupported task type '%T' for task '%s'", t, taskName)
 	}
 }
 
-func NewDoTaskRunner(taskList *model.TaskList, taskSupport TaskSupport) (*DoTaskRunner, error) {
+func NewDoTaskRunner(taskList *model.TaskList, workflowDef *model.Workflow) (*DoTaskRunner, error) {
 	return &DoTaskRunner{
 		TaskList:    taskList,
-		TaskSupport: taskSupport,
+		WorkflowDef: workflowDef,
 	}, nil
 }
 
 type DoTaskRunner struct {
 	TaskList    *model.TaskList
-	TaskSupport TaskSupport
+	WorkflowDef *model.Workflow
```

Comment:
This is unnecessary since the `WorkflowDev` is only used during run and not at the boostrap. When running, we have access to `TaskSupport.GetWorkflowDef`.